### PR TITLE
92 feature imix callback and main loop

### LINKED ISSUE #92: [feature] imix callback and main loop
**Is your feature request related to a problem? Please describe.**
Implement the main loop and callback mechanism for the imix agent.

**Describe the solution you'd like**
1. Web request, send task updates and claim new tasks (graphql)
2. Parse response
3. Web request to get new times
4. Untargz and run times

**Describe alternatives you've considered**
N/a

**Additional context**
N/a

### PR DESCRIPTION
**Pull request by hulto**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
Implements main_loop and callbacks for imix.

#### Which issue(s) this PR fixes:
Fixes #92

### CODE REVIEW

File: implants/imix/src/graphql.rs [hunk 1]
```diff
@@ -1,63 +1,391 @@
...
+        },
+    }) {
+        Ok(json_req_body) => json_req_body,
+        Err(error) => return Err(anyhow::anyhow!("Failed encode request to JSON\n{}", error)),
+    };
+
+    let client = reqwest::Client::new();
+
+    let response_text = match client.post(uri)
+    .header("Content-Type", "application/json")
```
Comment (hulto) [inline implants/imix/src/graphql.rs:128]: test https and https self signed cert.

**Review by hulto — COMMENTED** [review]
Comments with Nick.

**Comment by hulto** [comment]
Resolved all issues @Cictrone

**Review by Cictrone — APPROVED** [review]
LGTM

**Pull request merged**